### PR TITLE
Add saturate floating point builtin function

### PR DESCRIPTION
**Pull request by dneto0**
Fixes: #3012

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -8844,6 +8844,7 @@ value with the same sign.
   <tr><td>`sign(x)`<td colspan=2 style="text-align:left;">Correctly rounded
   <tr><td>`sin(x)`<td>Absolute error &le; 2<sup>-11</sup> inside the range [-&pi;, &pi;]<td>Absolute error &le; 2<sup>-7</sup> inside the range [-&pi;, &pi;]
   <tr><td>`sinh(x)`<td colspan=2 style="text-align:left;">Inherited from `(exp(x) - exp(-x)) * 0.5`
+  <tr><td>`saturate(x)`<td colspan=2 style="text-align:left;">Correctly rounded
```
Comment (alan-baker) [inline wgsl/index.bs:8847]: I think this is right, but it stands in contrast to the description of clamp. Should clamp say correctly rounded (modulo the given implementation)?

Comment (dneto0) [reply wgsl/index.bs:8847]: Yeah it's interesting.
@zoddicus proposed  using the "inherited from" wording for clamp.  In #2918 

You're right though, we can probably use "correctly rounded" there too.   Like with ldexp (which is a clearer case), the accuracy of a function can be strictly given even if there is freedom in how the function is computed.


[In Vulkan](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#spirvenv-precision-operation),   min, max, and clamp are all "correctly rounded".

Comment (dneto0) [reply wgsl/index.bs:8849]: Updated this PR to state Clamp's accuracy as correctly rounded, but that the infinitely precise results depends on the formulation used.

**Comment by dneto0** [comment]
Rebased and resolved conflicts

**Review by alan-baker — APPROVED** [review]

**Pull request merged**